File: apps/client/app/features/notifications/types/Notification.ts
export type NotificationType = 'organization.invitation';

export type NotificationItem = {
  id: string;
  type: NotificationType;
  organization_id: string;
  organization_name: string;
  organization_auth_state: 'member' | 'manager';
  created_at: string;
};

export type NotificationListResponse = {
  items: NotificationItem[];
};

File: apps/shared/schemas/notification.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class NotificationItemResponse(BaseModel):
    id: str
    type: str
    organization_id: UUID
    organization_name: str
    organization_auth_state: str
    created_at: datetime


class NotificationListResponse(BaseModel):
    items: list[NotificationItemResponse]
